### DISCUSSION TITLE
Statically generate site without duplicated page-data files

### DISCUSSION THREAD

**Discussion by stiaanj** [gatsbyjs/gatsby#33679] [Help]
I am trying to reduce my build size and removing unnecessary files from my public folder. We use Gatsby as a static site generator to build many unique sites based on the content they provide.

After digging through the public folder, I have noticed that the page-data.json files are still generated even if the content is already included HTML files. This is increasing the build size by 34.61%.

My question: How can I disable the generation of page-data and purely generate the files statically?

**Comment by LekoArts** [comment]
Hi!

You'll need to keep the `page-data.json` files around as they are necessary for client-side routing. When a user visits a page directly, they get served the HTML file. If they click on a link then the React app will only load the JSON file and do the routing on the client.

The size itself shouldn't matter as they won't be regenerated unless content changes (and you keep the cache around, e.g. this happens automatically on Gatsby Cloud).

And you can't disable the generation of those files.

**Reply by stiaanj** [reply]
Thank you for the quick response, that makes sense. If we purely wanted HTML files to be generated without the client-side routing, would that be possible? Our single build includes multiple region specific and languages specific pages. Depending on your region and language, a single content page will be displayed to you. The user never navigates to any other client routes.

**Reply by LekoArts** [reply]
No, we don't expose that functionality. With [incremental builds](https://www.gatsbyjs.com/docs/reference/release-notes/v3.0/#incremental-builds-in-oss) you would only build these once and then most probably keep them around as they wouldn't change.

Why is build size a concern for you?

**Reply by stiaanj** [reply]
The short answer is volume. We generate multiple independent builds on a daily basis and optimising the build size will reduce the deployment time and storage costs.
